xwiki-platform-distribution/xwiki-platform-distribution-flavor/xwiki-platform-distribution-flavor-test/xwiki-platform-distribution-flavor-test-ui/src/test/it/org/xwiki/test/ui/RegisterTest.java

```diff
@@ -97,7 +93,8 @@ public void testRegisterJohnSmith()
     })
     public void testRegisterExistingUser()
     {
-        registrationPage.fillRegisterForm(null, null, "Admin", null, null, null);
+        // Uses the empty string instead of the null value to empty the form fields (the null value just keep the value filled from the previously run test).
```

Comment:
This line is a bit too long. Do you know why the null values are ignored? Maybe  the text framework should handle them as empty strings?

Reply:
Currently `AbstractRegistrationPage#fillRegisterForm` is just skipping the field when the corresponding parameter is `null`.
For instance:
```java
if (firstName != null) {
  map.put("register_first_name", firstName);
}
// [...]
getForm().fillFieldsByName(map);
```
It seems to be done this way on purpose, but looking at the way this method is used, it shouldn't be too risky to consider the `null` value as the empty string for instance.

Reply:
OK, so ``null`` could be useful to keep the value unchanged because due to the way the method was designed you need to pass some values. E.g. you may want to change only the first name but not the rest when calling this method. Would be good to have this in the JavaDoc of ``fillRegisterForm``.